src/rgw/rgw_lc.cc

```diff
@@ -90,8 +90,14 @@ bool RGWLifecycleConfiguration::_add_rule(LCRule *rule)
   if (rule->get_filter().has_tags()){
     op.obj_tags = rule->get_filter().get_tags();
   }
-  auto ret = prefix_map.emplace(std::move(prefix), std::move(op));
-  return ret.second;
+
+  /* prefix is optional, update prefix map only if prefix...exists */
+  if (!prefix.empty()) {
+    auto ret = prefix_map.emplace(std::move(prefix), std::move(op));
+    return ret.second;
+  } else {
+    return true; /* suspicious, but */
```

Comment:
We could return true outside of the if body as well

Reply:
I think this should be covered with the above check, as earlier we used to reject an empty prefix, now the above check should handle only non empty prefixes.

Reply:
@theanalyst thanks for above comment, will resend shortly